[APPSEC-57] Modify rule S5146: Educational content (Commons) (#1132)
SonarSource/rspec: rules/S5146/impact.adoc

=== What is the potential impact?

> A web application redirects users to specific URLs based on user-controllable data. An attacker can manipulate a user to perform an HTTP request to the vulnerable web application. The HTTP request contains maliciously injected data that results in the user being redirected to an attacker's controlled evil domain.

Reviewer: I just realized that this is a repetition of what's already said in the rationale, e.g.:
```
An attacker with malicious intent could manipulate a user to browse into a specially crafted URL, such as ``++https://example.com?url=evil.com++``, to redirect the victim to his evil domain.
```

So we can just leverage this text and say:

```suggestion
Once the attackers trick a user into opening the malicious link, the user is redirected to a malicious domain controlled by the attacker. +
From then on, the attacker can perform various malicious actions on the victim's browser, ranging from entirely harmless to extreme organizational impact.
```

Author: ```suggestion
A web application redirects users to specific URLs based on user-controllable data and an attacker is able to trick a user into submitting a domain controlled by him, thus forcing the user to be redirected.

From then on, the attacker can perform various malicious actions on the victim's browser, some more impactful than others.
```

> Below are some real-world scenarios that illustrate the various effects of an attacker exploiting the vulnerability.

Author: ```suggestion
Below are some real-world scenarios that illustrate some impacts of an attacker exploiting the vulnerability.
```

==== Domain Mirroring

> A malicious link redirects to an attacker's controlled website mirroring the interface of a web application trusted by the user. Due to the similarity in the application appearance and the apparently trustable clicked hyperlink, the user struggles to identify that they are browsing on a malicious domain. Depending on the attacker's purpose, the malicious website can leak credentials, bypass Multi-Factor Authentication (MFA), and reach any authenticated data or action.

Reviewer: ```suggestion
A malicious link redirects to an attacker's controlled website mirroring the interface of a web application trusted by the user. Due to the similarity in the application appearance and the apparently trustable clicked hyperlink, the user struggles to identify that they are browsing on a malicious domain. +
Depending on the attacker's purpose, the malicious website can leak credentials, bypass Multi-Factor Authentication (MFA), and reach any authenticated data or action.
```

Hard line break suggestion to make the reading breathable

==== Malware Distribution

> A malicious link redirects to an attacker's controlled website that serves malware. On the same basis as the domain mirroring exploitation, the attacker develops a spearphishing or phishing campaign with a carefully crafted pretext that would result in the download and potential execution of a hosted malicious file. The worst-case scenario could result in complete system compromise.

Reviewer: ```suggestion
A malicious link redirects to an attacker's controlled website that serves malware. On the same basis as the domain mirroring exploitation, the attacker develops a spearphishing or phishing campaign with a carefully crafted pretext that would result in the download and potential execution of a hosted malicious file. +
The worst-case scenario could result in complete system compromise.
```
hard line break to emphasize the last line

=== What is the potential impact?

> A web application redirects users to specific URLs based on user-controllable data. An attacker is able to manipulate a user to perform an HTTP request to the vulnerable web application. The HTTP request contains maliciously injected data that would result in the user being redirected to an attacker's controlled evil domain.

Author: ```suggestion
A web application redirects users to specific URLs based on user-controllable data. An attacker can manipulate a user to perform an HTTP request to the vulnerable web application. The HTTP request contains maliciously injected data that would result in the user being redirected to an attacker's controlled evil domain.
```

> Tricking users into sending the manipulated HTTP request is usually the main task of exploiting an open redirection. Often, it requires an attacker to build a credible pretext to prevent suspicions from the victim. Attackers commonly use open redirect exploits in mass phishing campaigns.
> 

Author: ```suggestion
```

> Manipulating the user into performing the HTTP request is usually the main task of exploiting an open redirection as, more often than not, requires an attacker to build a credible pretext to prevent any suspicions arising from the victim. Threat actors commonly use open redirect exploits in mass phishing campaigns. These campaigns commonly occur when threat actors are in posession of leaked customer details for a specific entity which allows them to craft efficient and credible email templates. Banks and e-commerces are highly targeted. 

Author: ```suggestion
Manipulating the user into performing the HTTP request is usually the main task of exploiting an open redirection as, more often than not, requires an attacker to build a credible pretext to prevent any suspicions arising from the victim. Attackers commonly use open redirect exploits in mass phishing campaigns.
```

> Manipulating the user into performing the HTTP request is usually the main task of exploiting an open redirection as, more often than not, requires an attacker to build a credible pretext to prevent any suspicions arising from the victim. Threat actors commonly use open redirect exploits in mass phishing campaigns. These campaigns commonly occur when threat actors are in posession of leaked customer details for a specific entity which allows them to craft efficient and credible email templates. Banks and e-commerces are highly targeted. 

Author: ```suggestion
Tricking the user into performing the manipulated HTTP request is usually the main task of exploiting an open redirection. Often, it requires an attacker to build a credible pretext to prevent suspicions from the victim. Attackers commonly use open redirect exploits in mass phishing campaigns.
```

Below are some real-world scenarios that illustrate the various effects of an attacker exploiting the vulnerability.

==== Domain Mirroring

> Redirection to an attacker's controlled website mirroring the interface of a web application trusted by the user. Due to the similarity in the application appearance and the apparently trustable clicked hyperlink, the user struggles to identify that is browsing on a malicious domain. Credentials can be leaked, MFAs can be bypassed and any authenticated data or action can be reached by the attacker.

Author: ```suggestion
A malicious link redirects to an attacker's controlled website mirroring the interface of a web application trusted by the user. Due to the similarity in the application appearance and the apparently trustable clicked hyperlink, the user struggles to identify that they are browsing on a malicious domain. Depending on the attacker's purpose, the malicious website can leak credentials, bypass Multi-Factor Authentication (MFA), and reach any authenticated data or action.
```

Applied suggestions with two minor change.

==== Malware Distribution

> User redirection to an attackers' controlled website which serves malicious files. On the same basis as the domain mirroring exploitation, the attacker develops a spearphishing or phishing campaign with a carefully crafted pretext that would result in the download and potential execution of a hosted malicious file. On the worst could result in full system compromise.

Author: ```suggestion
A malicious link redirects to an attacker's controlled website, which serves malware. On the same basis as the domain mirroring exploitation, the attacker develops a spearphishing or phishing campaign with a carefully crafted pretext that would result in the download and potential execution of a hosted malicious file. The worst could result in complete system compromise.
```
